src/backend/distributed/utils/connection_cache.c

```diff
@@ -234,9 +228,31 @@ ReportRemoteError(PGconn *connection, PGresult *result)
 		}
 	}
 
-	ereport(WARNING, (errcode(sqlState),
-					  errmsg("%s %s:%s", errorPrefix, nodeName, nodePort),
-					  errdetail("Remote message: %s", remoteMessage)));
+	/* use diffent warning messages for different error types */
+	if (sqlState == ERRCODE_CONNECTION_FAILURE)
+	{
+		ereport(WARNING, (errcode(sqlState),
+						  errmsg("connection failed to %s:%s", nodeName, nodePort),
+						  errdetail("%s", remoteMessage)));
+	}
+	else
+	{
+		if (remoteDetail == NULL)
+		{
+			ereport(WARNING, (errcode(sqlState),
+							  errmsg("remote message: %s", remoteMessage),
+							  errhint("Bad result from %s:%s.", nodeName, nodePort)));
```

Comment:
"HINT: Bad result" feels a bit confusing, maybe if we have something like:

    erggmsg("remote message from %s:%s: %s", nodeName, nodePort, remoteMessage)

we can omit the hint.